cluster-autoscaler/scale_down.go

```diff
@@ -163,6 +164,31 @@ func ScaleDown(
 		return ScaleDownNoUnneeded, nil
 	}
 
+	emptyNodes := filterEmptyNodes(simulator.FindEmptyNodesToRemove(candidates, pods), maxEmptyBulkDelete, cloudProvider)
+	if len(emptyNodes) > 0 {
+
+		confirmation := make(chan error, len(emptyNodes))
+		for _, node := range emptyNodes {
+			glog.V(0).Infof("Scale-down: removing empty node %s", node.Name)
+			simulator.RemoveNodeFromTracker(usageTracker, node.Name, unneededNodes)
+			go func(nodeToDelete *kube_api.Node) {
+				confirmation <- deleteNodeFromCloudProvider(nodeToDelete, cloudProvider, recorder)
+			}(node)
+		}
+		var finalError error
+		for _ = range emptyNodes {
+			if err := <-confirmation; err != nil {
+				glog.Errorf("Problem with empty node deletion: %v", err)
+				finalError = err
+			}
+		}
+		if finalError == nil {
+			return ScaleDownNodeDeleted, nil
+		} else {
+			return ScaleDownError, fmt.Errorf("failed to delete at least one empty node: %v", finalError)
```

Comment:
The result is only important from the timing perspective. If we failed to delete some node we wait a bit more when trying to delete something more.